cnxdb/migrations/20170911201035_transform_cnxml_to_html.py

```diff
@@ -133,6 +157,27 @@ def up(cursor):
                                             time.ctime(est_complete),
                                             timedelta(0, remaining_est)))
 
+    # index.cnxml.html must be transformed first before baking as baking uses
+    # index.cnxml.html
+    for batch in _batcher(baked_to_transform, batch_size):
+        book_module_idents = tuple([i[0] for i in batch])
+        logger.debug('Rebaking books {}'.format(book_module_idents))
+
+        cursor.execute("""\
+UPDATE modules SET stateid = (
+    SELECT stateid FROM modulestates WHERE statename = 'post-publication'
+) WHERE module_ident IN %s""", (book_module_idents,))
+
+        num_complete += len(batch)
+        percent_comp = num_complete * 100.0 / num_todo
+        elapsed = time.time() - start
+        remaining_est = elapsed * (num_todo - num_complete) / num_complete
+        est_complete = start + elapsed + remaining_est
+        logger.info('{:.1f}% complete '
+                    'est: "{}" ({})'.format(percent_comp,
+                                            time.ctime(est_complete),
+                                            timedelta(0, remaining_est)))
```

Comment:
This should probably be refactored so that we aren't copying code. I know I've copied this block of code into another migration even. 😖

Reply:
Oh... yeah, definitely :sweat_smile:

Reply:
I'll leave this for next time...

Reply:
copying code 😮

Reply:
😱 I mean no problem from my side 😉. Copying is sometimes necessary and I'm for releasing early rather than have perfect non copied code (which fulfills the task perfectly).